Tests/SentryTests/Integrations/SessionReplay/SentryBreadcrumbReplayConverterTests.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+@testable import Sentry
+import XCTest
+
+class SentryBreadcrumbReplayConverterTests: XCTestCase {
+    func testReplayBreadcrumbsWithEmptyArray() {
+        let sut = SentryBreadcrumbReplayConverter()
+        let result = sut.replayBreadcrumbs(from: [])
+        XCTAssertTrue(result.isEmpty)
+    }
+    
+    func testReplayBreadcrumbWithNilTimestamp() {
+        let sut = SentryBreadcrumbReplayConverter()
+        let breadcrumb = Breadcrumb(level: .debug, category: "Breadcrumb")
+        breadcrumb.timestamp = nil
+        let result = sut.replayBreadcrumbs(from: [breadcrumb])
+        XCTAssertEqual(result.count, 0)
+    }
+    
+    func testNavigationBreadcrumbAppLifecycle() {
+        let sut = SentryBreadcrumbReplayConverter()
+        let crumb = Breadcrumb(level: .info, category: "app.lifecycle")
+        crumb.type = "navigation"
+        crumb.data = ["state": "foreground"]
+        let result = sut.replayBreadcrumbs(from: [crumb])
+        
+        XCTAssertEqual(result.count, 1)
+        let event = result.first?.serialize()
+        let eventData = event?["data"] as? [String: Any]
+        let eventPayload = eventData?["payload"] as? [String: Any]
+        
+        XCTAssertEqual(event?["type"] as? Int, 5)
```

Comment:
`l`: Isn't that a bit cleaner? The same applies to all other event["type"] assertions.
```suggestion
        XCTAssertEqual(event?["type"] as? Int, SentryRRWebEventType.custom.rawValue)
```

Reply:
Its clear, but we can change `SentryRRWebEventType.custom` by mistake and the test will succeed where the server side is expecting 5.